merlin/models/tf/predictions/dot_product.py

```diff
@@ -0,0 +1,371 @@
+           by default None
+       pre: Optional[Block], optional
+           Optional block to transform predictions before applying the prediction layer,
+           by default None
+       post: Optional[Block], optional
+           Optional block to transform predictions after applying the prediction layer,
+           by default None
+       logits_temperature: float, optional
+           Parameter used to reduce model overconfidence, so that logits / T.
+           by default 1.
+       name: Optional[Text], optional
+           Task name, by default None
+       default_loss: Union[str, tf.keras.losses.Loss]
+           Default loss to set if the user does not specify one
+       default_metrics: Sequence[tf.keras.metrics.Metric]
+           Default metrics to set if the user does not specify any
+       query_name : str, optional
+           Identify query tower for query/user embeddings, by default 'query'
+       item_name : str, optional
+           Identify item tower for item embeddings, by default'item'
+    """
+
+    DEFAULT_K = 10
+
+    def __init__(
+        self,
+        schema: Schema,
+        negative_samplers: ItemSamplersType = "in-batch",
+        downscore_false_negatives=False,
+        target: Optional[str] = None,
+        pre: Optional[Layer] = None,
+        post: Optional[Layer] = None,
+        logits_temperature: float = 1.0,
+        name: Optional[str] = None,
+        default_loss: Union[str, tf.keras.losses.Loss] = "categorical_crossentropy",
+        default_metrics: Sequence[tf.keras.metrics.Metric] = (
+            RecallAt(DEFAULT_K),
+            MRRAt(DEFAULT_K),
+            NDCGAt(DEFAULT_K),
+            AvgPrecisionAt(DEFAULT_K),
+            PrecisionAt(DEFAULT_K),
+        ),
+        query_name: str = "query",
+        item_name: str = "item",
+        **kwargs,
+    ):
+        prediction = kwargs.pop("prediction", DotProduct(query_name, item_name))
+        prediction_with_negatives = kwargs.pop(
+            "prediction_with_negatives",
+            ContrastiveDotProduct(
+                schema,
+                negative_samplers,
+                downscore_false_negatives,
+                query_name=query_name,
+                item_name=item_name,
+            ),
+        )
+
+        super().__init__(
+            prediction=prediction,
+            prediction_with_negatives=prediction_with_negatives,
+            default_loss=default_loss,
+            default_metrics=default_metrics,
+            name=name,
+            target=target,
+            pre=pre,
+            post=post,
+            logits_temperature=logits_temperature,
+            **kwargs,
+        )
+
+    def compile(self, negative_sampling=None, downscore_false_negatives=False):
+        self.prediction_with_negatives.negative_sampling = negative_sampling
+        self.prediction_with_negatives.downscore_false_negatives = downscore_false_negatives
+
+    # TODO
```

Comment:
If this TODO is still relevant, can you add more context of what it refers to (given that the method seems to have an implementation

Reply:
Thanks for the catch!  I just removed the # TODO as the `add_sampler` is now implemented and tested.